frontend/src/metabase/static-viz/components/XYChart/XYChart.tsx

```diff
@@ -215,18 +231,37 @@ export const XYChart = ({
 
         <AxisBottom
           scale={xScale.scale}
-          label={areXTicksRotated ? undefined : settings.labels.bottom}
+          label={settings.labels.bottom}
           top={yMin}
           left={xMin}
           numTicks={xTicksCount}
-          labelOffset={xLabelOffset}
+          labelOffset={getXLabelOffset(
+            xTickDisplay,
+            style.axes.ticks.fontSize,
+            Math.max(
+              ...tickValues.map(tickValue =>
+                measureTextWithDefault(
+                  truncateXTickLabel(
+                    formatXTick(
+                      tickValue.valueOf(),
+                      settings.x.type,
+                      settings.x.format,
+                    ),
+                    xTickWidthLimit,
+                    style.value.fontSize,
+                  ),
+                ),
+              ),
+            ),
+          )}
           stroke={style.axes.color}
           tickStroke={style.axes.color}
-          hideTicks={settings.x.tick_display === "hide"}
+          hideTicks={areXTicksHidden}
           labelProps={labelProps}
           tickFormat={value =>
             formatXTick(value.valueOf(), settings.x.type, settings.x.format)
           }
+          tickValues={tickValues}
```

Comment:
I don't really need to pass this as `AxisBottom` already has a default value for this, but I explicitly pass this value because I want to point out that this is the same value used for calculating max x tick values width.

https://github.com/airbnb/visx/blob/00f5d14dae032ef15dc9c97ec9eebf508cadd265/packages/visx-axis/src/axis/Axis.tsx#L49